File: ebook-be/src/main/java/com/example/ebookbe/Service/UserService.java
package com.example.ebookbe.Service;

import com.example.ebookbe.Entity.User;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestParam;

import java.util.List;

public interface UserService {
    List<User> findAll();

    User getUserById(int id);

    List<User> getUsers();

    User changeStatus(int id);

    int checkLogin(String username, String password);

    boolean checkUsername(String username);

    User addUser(String username, String password, String name, String email, String avatar);

    User updateUser(String username, String password, String name, String email, String avatar);

    User findById(int id);

}

File: ebook-react/src/services/UserService.js
import {getRequest,postRequest} from "./Ajax";

export const login=(value,callback)=>{
    const url = "http://localhost:8080/login";
    postRequest(url,value,callback);
}

export const changeStatus=(id,callback)=>{
    const url = `http://localhost:8080/changestatus/${id}`;
    getRequest(url,callback);
}

export const getUsers=(callback)=>{
    const url = `http://localhost:8080/getusers`;
    getRequest(url,callback);
}

export const getUserById=(id,callback)=>{

    const url = `http://localhost:8080/getuserbyid/${id}`;
    getRequest(url, callback);
}

export const checkUsername=(username,callback)=>{
    const url = `http://localhost:8080/checkusername/${username}`;
    getRequest(url,callback);
}

export const addUser=(object,callback)=>{
    const url = `http://localhost:8080/adduser`;
    postRequest(url,object,callback);
}

export const updateUser=(object,callback)=>{
    const url = `http://localhost:8080/updateuser`;
    postRequest(url,object,callback);
}
